gazebo/rendering/Camera.cc

```diff
@@ -421,6 +422,8 @@ void Camera::SetScene(ScenePtr _scene)
 //////////////////////////////////////////////////
 void Camera::Update()
 {
+  IGN_PROFILE("rendering::Camera::Update");
+  IGN_PROFILE_BEGIN("Update");
```

Comment:
Just a note that `IGN_PROFILE` will profile the current scope, so doing an addition `IGN_PROFILE_BEGIN/IGN_PROFILE_END` is a duplication of work.

Reply:
Maybe the documentation needs to be updated to state that more clearly.

`IGN_PROFILE_BEGIN/IGN_PROFILE_END` should really only be required when you don't want to profile a full scope.

Reply:
It will also reduce your profiler "callstacks" quite a bit.

Reply:
I experimented some problems when I was not using `IGN_PROFILE_BEGIN/IGN_PROFILE_END`. Sometime an entire thread disappeared.

Reply:
Interesting.  In that case, the `IGN_PROFILE` call itself shouldn't be needed?

Reply:
As far as I remember from the other profilers PRs, the `IGN_PROFILE("")` calls are to define the method that it's being called. The other call `IGN_PROFILE_BEGIN` it's for a subsection is required (fill arrays, send, render, etc). Does it make sense?

Reply:
Some additional variables to experiment with to tune remotery: https://github.com/ignitionrobotics/ign-common/blob/512f077c7970813ed37d55d5f83111765cba9d4e/profiler/src/RemoteryProfilerImpl.hh#L43-L46

They correspond to similarly-named variables in the remotery configuration structure: https://github.com/Celtoys/Remotery/blob/master/lib/Remotery.h#L354-L397